app/models/game.rb

```diff
@@ -1,36 +1,40 @@
 class Game < ApplicationRecord
+  # belongs_to :black_player, class_name: 'Player'
   has_many :pieces
   has_many :players
-  scope :available, -> {where("total_players = 1")}
+  scope :available, -> { where('total_players = 1') }
   belongs_to :user
   after_create :current_user_is_white_player
   after_create :populate
 
+  # to initialize each game with the white_player as the user who created the game
+  # white_player_id needs to exist in the database
 
-  #to initialize each game with the white_player as the user who created the game 
-  #white_player_id needs to exist in the database
-  
-  #we need this for everything else to work
+  # we need this for everything else to work
   def square_occupied?(x_current, y_current)
-    pieces.active.where({x: x_current, y: y_current}).any? ? true : false
+    pieces.active.where(x: x_current, y: y_current).any? ? true : false
+  end
+
+  def add_black_player!(player)
+    self.black_player = player
```

Comment:
I'm not sure if this will work or if it needs to be `player.id`. In either case, a test needs to be added to the `game_spec` to make sure the method works as intended.

Reply:
according to my mentor when we made the changes it was working since the last pull.